pkg/tsdb/azuremonitor/azure-resource-graph-datasource.go

```diff
@@ -236,8 +236,45 @@ func (e *AzureResourceGraphDatasource) unmarshalResponse(res *http.Response) (Az
 	}()
 
 	if res.StatusCode/100 != 2 {
+		type ErrorObject struct {
+			Code    string `json:"code"`
+			Message string `json:"message"`
+			Details []struct {
+				Code                    string `json:"code"`
+				Message                 string `json:"message"`
+				Line                    int    `json:"line,omitempty"`
+				CharacterPositionInLine int    `json:"characterPositionInLine,omitempty"`
+				Token                   string `json:"token,omitempty"`
```

Comment:
For simplicity, `Code` and `Message` probably enough to compose an error message.
If user is able to look up the full JSON when needed, then details like `Line`, `Token` etc, are probably not needed for error message.

Reply:
Presence custom fields like `Line`, `Token`, `CharacterPositionInLine` etc. depends on the error `Code`.
Common fields are only `code`, `message`, details` and some others which are less important.

Reply:
Gotcha, thanks!